cassandra/cqltypes.py

```diff
@@ -291,7 +291,11 @@ def cass_parameterized_type_with(cls, subtypes, full=False):
             cname = apache_cassandra_type_prefix + cname
         if not subtypes:
             return cname
-        sublist = ', '.join(styp.cass_parameterized_type(full=full) for styp in subtypes)
+        fieldnames = getattr(cls, 'fieldnames', None)
+        if fieldnames and any(fieldnames):
```

Comment:
I'm guessing this is meant to only apply to DynamicCompositeType, but UserType also uses ``fieldnames``, so it seems like the wrong thing might happen here.